src/gui/editorwidgets/qgsdefaultsearchwidgetwrapper.cpp

```diff
@@ -96,6 +96,7 @@ void QgsDefaultSearchWidgetWrapper::initWidget( QWidget* widget )
   mContainer->layout()->addWidget( mLineEdit );
   mContainer->layout()->addWidget( mCheckbox );
   connect( mLineEdit, SIGNAL( textChanged( QString ) ), this, SLOT( setExpression( QString ) ) );
+  connect( mLineEdit, SIGNAL( returnPressed() ), this, SLOT( filterChanged() ) );
```

Comment:
Hint: you can directly connect signals to signals.
Just do:
    connect( mLineEdit, SIGNAL( returnPressed() ), this, SIGNAL( expressionChanged() ) );

That should keep the API smaller